processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -665,7 +672,9 @@ private void mergeIndexesAndWriteColumns(
         if (dimensionSkipFlag.get(i)) {
           continue;
         }
-        if (dims[i] == null || dims[i].length == 0 || (dims[i].length == 1 && dims[i][0] == 0)) {
+        if (dims[i] == null || dims[i].length == 0) {
```

Comment:
i think this logic is tricky so it is worth a comment here but what is happening